src/Analyzer.JsonRuleEngine.UnitTests/JsonRuleEngineTests.cs

```diff
@@ -277,57 +275,44 @@ private void AssertEvaluationsAndResultsAreAsExpected(IEvaluation evaluation, in
                 }
             }]", DisplayName = "Invalid regex pattern")]
         [ExpectedException(typeof(JsonRuleEngineException))]
-        public void EvaluateRules_InvalidRule_ExceptionIsThrown(string invalidRule)
+        public void Create_InvalidRules_ExceptionIsThrown(string invalidRule)
         {
-            // Arrange
-            var template =
-                JObject.Parse(
-                @"{
-                    ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#"",
-                    ""resources"": [
-                        {
-                            ""type"": ""Microsoft.ResourceProvider/resource0"",
-                            ""properties"": {
-                                ""somePath"": ""someValue"",
-                                ""someOtherPath"": ""someOtherValue""
-                            }
-                        }
-                    ]
-                }");
-
-            string expectedFileId = "MyTemplate";
-
-            TemplateContext templateContext = new TemplateContext
-            {
-                OriginalTemplate = template,
-                ExpandedTemplate = template,
-                IsMainTemplate = true,
-                TemplateIdentifier = expectedFileId
-            };
+            // Act
+            JsonRuleEngine.Create(invalidRule, t => null);
+        }
 
-            // Setup mock line number resolver
-            var mockLineResolver = new Mock<ILineNumberResolver>();
+        [ExpectedException(typeof(JsonRuleEngineException))]
+        public void Create_NullRules_ExceptionIsThrown()
```

Comment:
I'll leave it as `Create_...` based on my comment above.

Reply:
I know it's a little weird to give it the name of the private method that's not exposed, but I mention it since it's technically testing the logic of `ParseRuleDefinition`. I just thought it was a little confusing when I was reading the test and noticed the logic isn't actually in Create.

Reply:
From a developer point of view: yes, the logic is in a different function.
From a unit test point of view: how the logic is organized is abstracted from the test.  `Create` is treated as a black box.  This test is simply verifying "If the `Create` method is called with `null` for the rules, there should be an exception."

Another way to think of it - if the `ParseRuleDefinitions` function was broken up into multiple pieces called from inside `Create`, it wouldn't affect any logic from the unit tests, and assuming the refactoring is done correctly, all tests would pass as they were.  But if the names were based on where the logic was internally, then the test names would need to be updated even though all the code remains the same.  I think tests should only be updated if logic or method signatures change.

Reply:
However, going through this exercise has made me realize the null (and whitespace) checks should actually be in `Create` instead of in `ParseRuleDefinitions`, so I'll make that update.